### PR TITLE
Update the bundle provider state

### LINKED ISSUE #115: bundle `uint` vs `uint[]`
This change makes some sense here,
but it will add more gas to use
this will allow us to be more flexible but at the cost of some gas store

this is good for a bundle factory:
we send NFT's to the factory (with the same token)
and click on build - which will give us the Total NFT

in the case of the factory, we do no need to make them in order.

_Originally posted by @Lomet in https://github.com/The-Poolz/LockDealNFT/pull/114#discussion_r1237067524_

### PR DESCRIPTION
**Pull request by 0xSolcroot**
close #115

### CODE REVIEW

**Comment by Lomet** [comment]
Please check https://github.com/The-Poolz/LockDealNFT/pull/135/files

File: contracts/LockDealNFT/LockDealNFT.sol [hunk 1]
```diff
@@ -94,15 +94,15 @@ contract LockDealNFT is LockDealNFTModifiers, ILockDealNFTEvents {
...
-        if(poolIdToVaultId[poolId] == 0 && !approvedProviders[msg.sender]) {
-            newOwner = provider;
-            dataPoolId = poolId - 2;
-        }
-        else {
-            dataPoolId = poolId;
-        }
+        uint256 dataPoolId = poolId;        
+        // // refund provider case
+        // if(poolIdToVaultId[poolId] == 0 && !approvedProviders[msg.sender]) {
```
Comment (YouStillAlive) [inline contracts/LockDealNFT/LockDealNFT.sol:99]: In this case, the `LockDealBundleProvider` contract does not match the conditions.
```solidity
if(!approvedProviders[msg.sender])
{
       // LockDealBundleProvider can't be here
}
```

But in any case, the next version of the pefund provider should change the if logic.

Comment (0xSolcroot) [reply contracts/LockDealNFT/LockDealNFT.sol:99]: The`split` function on LockDealNFT is called by the user, so `LockDealBundleProvider` contract match the condition.
Workflow: User -> Split(LockDealNFT) -> Split(LockDealBundleProvider)
Here is the test you can take a look at.
https://github.com/The-Poolz/LockDealNFT/blob/6b6172f432e9d45c44e3e852409caa6684acb24b/test/LockDealProvider.ts#L83
So if you can add any specific condition for the refund provider, that would be helpful.

Comment (YouStillAlive) [reply contracts/LockDealNFT/LockDealNFT.sol:99]: compromise solution: https://github.com/The-Poolz/LockDealNFT/pull/138

**Review by Lomet — APPROVED** [review]
Looks good !

**Review by YouStillAlive — APPROVED** [review]
:thumbsup:

**Pull request merged**